### ISSUE TITLE
feat: provide more information for debugging dex changes

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#2226]
When users get a warning about dex changes they sometimes don't know why.

We could probably do more to show them what changed.  There exist "dex diffing" programs afaict.

**Comment by erickzanardo** [comment]
Do we have any easy way to reproduce dex changes?

**Comment by eseidel** [comment]
https://github.com/sisbell/dexd is another (old) attempt at this.

**Comment by eseidel** [comment]
https://github.com/linkedin/dex-test-parser https://github.com/bunseokbot/dexparser https://crates.io/crates/dexparser are all parser implementations.

Honestly it looks a lot like a dart snapshot, except documented, so probably not very hard to parse.  We already have an LEB128 parser.
